File: Examples/CSharp/DrawingObjects/Comments/AddImageToComment.cs
using System.IO;

using Aspose.Cells;
using System.Drawing;

namespace Aspose.Cells.Examples.DrawingObjects.Comments
{
    public class AddImageToComment
    {
        public static void Main(string[] args)
        {
            // The path to the documents directory.
            string dataDir = Aspose.Cells.Examples.Utils.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType);

            // Create directory if it is not already present.
            bool IsExists = System.IO.Directory.Exists(dataDir);
            if (!IsExists)
                System.IO.Directory.CreateDirectory(dataDir);

            //Instantiate a Workbook
            Workbook workbook = new Workbook();

            //Get a reference of comments collection with the first sheet
            CommentCollection comments = workbook.Worksheets[0].Comments;

            //Add a comment to cell A1
            int commentIndex = comments.Add(0, 0);
            Comment comment = comments[commentIndex];
            comment.Note = "First note.";
            comment.Font.Name = "Times New Roman";

            //Load an image into stream
            Bitmap bmp = new Bitmap(dataDir + "logo.jpg");
            MemoryStream ms = new MemoryStream();
            bmp.Save(ms, System.Drawing.Imaging.ImageFormat.Png);

            //Set image data to the shape associated with the comment
            comment.CommentShape.FillFormat.ImageData = ms.ToArray();

            //Save the workbook
            workbook.Save(dataDir + "book1.out.xlsx", Aspose.Cells.SaveFormat.Xlsx);

            
        }
    }
}

File: Examples/VisualBasic/DrawingObjects/Comments/AddImageToComment.vb
Imports Microsoft.VisualBasic
Imports System.IO

Imports Aspose.Cells
Imports System.Drawing

Namespace DrawingObjects.Comments
    Public Class AddImageToComment
        Public Shared Sub Run()
            ' ExStart:1
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType)

            ' Create directory if it is not already present.
            Dim IsExists As Boolean = System.IO.Directory.Exists(dataDir)
            If (Not IsExists) Then
                System.IO.Directory.CreateDirectory(dataDir)
            End If

            ' Instantiate a Workbook
            Dim workbook As New Workbook()

            ' Get a reference of comments collection with the first sheet
            Dim comments As CommentCollection = workbook.Worksheets(0).Comments

            ' Add a comment to cell A1
            Dim commentIndex As Integer = comments.Add(0, 0)
            Dim comment As Comment = comments(commentIndex)
            comment.Note = "First note."
            comment.Font.Name = "Times New Roman"

            ' Load an image into stream
            Dim bmp As New Bitmap(dataDir & "logo.jpg")
            Dim ms As New MemoryStream()
            bmp.Save(ms, System.Drawing.Imaging.ImageFormat.Png)

            ' Set image data to the shape associated with the comment
            comment.CommentShape.Fill.ImageData = ms.ToArray()

            ' Save the workbook
            workbook.Save(dataDir & "output.xlsx", Global.Aspose.Cells.SaveFormat.Xlsx)
            ' ExEnd:1

        End Sub
    End Class
End Namespace